include/boost/geometry/formulas/vertex_longitude.hpp

```diff
@@ -0,0 +1,174 @@
+// Boost.Geometry
+
+// Copyright (c) 2016 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_MAXIMUM_LONGITUDE_HPP
+#define BOOST_GEOMETRY_FORMULAS_MAXIMUM_LONGITUDE_HPP
+
+#include <boost/geometry/formulas/spherical.hpp>
+#include <boost/geometry/algorithms/detail/flattening.hpp>
+#include <boost/geometry/core/srs.hpp>
+#include <boost/mpl/assert.hpp>
+
+#include <boost/math/special_functions/hypot.hpp>
+
+namespace boost { namespace geometry { namespace formula
+{
+
+/*!
+\brief Algorithm to compute the vertex longitude of a geodesic segment. Vertex is
+a point on the geodesic that maximizes (or minimizes) the latitude. The algorithm
+is given the vertex latitude.
+*/
+
+template <typename CT>
+class vertex_longitude_on_sphere
+{
+
+public:
+
+    template <typename T>
+    static inline CT apply(T const lat1,
+                           T const lat2,
+                           T const lat3,
+                           T const dlon)
+    {
```

Comment:
I think (here and in other places) this should be T const&, for user defined numeric types.